text/0000-create-stackify-repo.md

```diff
@@ -0,0 +1,141 @@
+
+
+# What it is
+[what-it-is]: #what-it-is
+Stackify will be a tool that will be useful to users who would like to customize the published stack images or turn their own base images into CNB compliant stack images.
+
+Users can use stackify to add packages and ca-certificates to a base image as well as add all required CNB metadata to the image.
+
+Stackify can be integrated into Pack in the form of a `pack create-stack` command.
+
+# How it Works
+[how-it-works]: #how-it-works
+For stacks defined by the buildpacks project, Stackify will add mixins labels for each package installed on the image. If Stackify is used to give add a stack ID other than one defined by the buildpacks project, mixin labels will not be added.
+
+If Stackify is used with an already valid stack image, the `CNB_*` labels and users will not be changed unless new user and group ID's are passed, in which case the labels and user will be overwritten.
+
+If run on the image being "stackify-ed", Stackify can be used to add packages and ca-certificates.
+
+Stackify will use Kaniko to create additional image layers with the additional packages and ca-certificates that users choose to add to their base image. It will also use GGCR to manipulate the image metadata.
+
+
+### Proposed Interface
+Stackify will be run once for each image (build and run):
+```
+stackify 
+  --base-image string
+    	Source image ref
+  --stack-image string
+    	Destination for stack image
+  --certs-dir string
+    	Path to directory with CA Certs to add to image
+  --group-id string
+    	CNB Group ID
+  --image-type string
+    	Type of image to create (build or run)
+  --mixins string
+    	Comma-separated list of mixins to add to labels
+  --packages string
+    	Comma-separated list of packages to add to image
+  --stack-id string
+      Stack ID to add to image metadata
+  --user-id string
+    	CNB User ID
+```
+
+This will be called under the hood by the `pack create-stack` command, which will take a TOML config file:
+
+`pack create-stack my-stack -c /path/to/config.toml`
+
+Config file:
+```
+[build]
+image = "paketobuildpacks/build:base"
+outputTag = "some-registry/some-custom-stack-build-image"
+
+[run]
+image = "paketobuildpacks/run:base"
+outputTag = "some-registry/some-custom-stack-run-image"
+
+[caCerts]
+
+[caCerts.build]
+path = "/path/to/dir/with/build-certs"
+
+[caCerts.run]
+path = "/path/to/dir/with/run-certs"
+
+[[packages]]
+name = "cowsay"
+
+[[packages]]
+name = "fortune"
+phase = "build"
+
+[[packages]]
+name = "rolldice"
+phase = "run"
+
+[[mixins]]
+name = "set=build-utils"
+phase = "build"
+
+[[mixins]]
+name = "set=run-utils"
+phase = "run"
+
+[[mixins]]
+name = "set=shared-utils"
+
+[user]
+userID = 1000
+groupID = 1000
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+* Stackify must be run on the image being "stackify-ed" in order to add ca-certificates and packages.
```

Comment:
To this point: are there any requirements on the image, in order for stackify to be run within it? Can it be run on a totally scratch image? A windows container?

Reply:
On day one, it would only support bionic images but we would hope to expand that later on.